### PR TITLE
chore: integration tests using gherkin test suite

### PR DESCRIPTION
**Pull request by skyerus**
## This PR

- integration with [the gherkin test suite](https://github.com/open-feature/test-harness)
- runs gherkin suite as part of the PR checks

### Related Issues

Fixes https://github.com/open-feature/go-sdk/issues/88

### CODE REVIEW

**Review by beeme1mr — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -4,6 +4,8 @@ mockgen:
 	mockgen -source=pkg/openfeature/hooks.go -destination=pkg/openfeature/hooks_mock_test.go -package=openfeature
 	mockgen -source=pkg/openfeature/mutex.go -destination=pkg/openfeature/mutex_mock_test.go -package=openfeature
 test:
+	go test --short -cover ./...
+integration-test: # dependent on `docker run -p 8013:8013 ghcr.io/open-feature/flagd-testbed:latest`
```
Comment (toddbaert) [inline Makefile:8]: Not necessary, but in the Java and JS repos, I added the git submodule retrieval into the integration test goal/script. This has the benefit of "just working" when somebody runs it locally.

File: integration/evaluation_test.go [hunk 2]
```diff
@@ -0,0 +1,658 @@
...
+) (context.Context, error) {
+	got, err := client.StringValue(ctx, flagKey, defaultValue, openfeature.EvaluationContext{})
+	if err != nil {
+		return ctx, fmt.Errorf("openfeature client: %w", err)
+	}
+
+	return context.WithValue(ctx, ctxStorageKey{}, got), nil
+}
+
+func theResolvedStringValueShouldBe(ctx context.Context, expectedValue string) error {
```
Comment (toddbaert) [inline integration/evaluation_test.go:79]: There's a few functions here like this one and `theResolvedIntegerValueShouldBe` that might be able to be made generic so they can work for multiple gherkin tests. Probably not possible with the boolean version wherein you'll need to parse.

This is entirely not-necessary though.

**Comment by toddbaert** [comment]
Added minor comments. Works (and fails) as expected for me locally. Nice job, this adds a lot of confidence.

**Review by toddbaert — APPROVED** [review]
Added minor comments. Works (and fails) as expected for me locally. Nice job, this adds a lot of confidence.

File: integration/evaluation_test.go (hunk 2 above)
Comment (skyerus) [reply integration/evaluation_test.go:79]: Do you mean locating any functions where behaviour is being duplicated and merging them into a singular function?
I can't see any instances where this is happening. `theResolvedStringValueShouldBe` and `theResolvedStringResponseShouldBe` are similar in what they're asserting but the execution is different due to the way the data is store/retrieved.

Maybe I've misunderstood.

File: Makefile (hunk 1 above)
Comment (skyerus) [reply Makefile:8]: Nice suggestion, done

File: integration/evaluation_test.go (hunk 2 above)
Comment (toddbaert) [reply integration/evaluation_test.go:79]: I mean functions like `theResolvedStringValueShouldBe` and `theResolvedIntegerValueShouldBe`. These functions are the same, except the interface casting and string format token. which are associated with the type of the `expectedValue`. That's why I think (possibly incorrectly) they could be made generic.

This is absolutely not a necessary change though, even if I'm correct.

Comment (skyerus) [reply integration/evaluation_test.go:79]: I think due to the way we're storing the typed data and retrieving it for the assertion the complexity of doing this wouldn't be worth the benefits

**Pull request merged**